upstairs/src/lib.rs

```diff
@@ -2248,20 +2254,10 @@ struct Downstairs {
 }
 
 impl Downstairs {
-    fn new(target: Vec<SocketAddr>, log: Logger) -> Self {
-        // Fill the repair hashmap based on the
-        // addresses from each downstairs.
-        let mut ds_repair = HashMap::new();
-        for (i, addr) in target.iter().enumerate() {
-            assert!(addr.port() < u16::MAX - REPAIR_PORT_OFFSET);
-            let port = addr.port() + REPAIR_PORT_OFFSET;
-            let repair_addr = SocketAddr::new(addr.ip(), port);
-            ds_repair.insert(i as u8, repair_addr);
-        }
-
+    fn new(log: Logger) -> Self {
         Self {
             ds_uuid: HashMap::new(),
-            ds_repair,
+            ds_repair: HashMap::new(),
```

Comment:
Not for this PR, but man this seems ripe for a refactor.
Every element in this is a tuple or hash map of the three downstairs.  It seems like there could be a better way to do this.  Some are index based, and some by a hash key, it's.. it's not my proudest moment.